contracts/TokenDistributor.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import {StakingRewardsV2} from "./StakingRewardsV2.sol";
+import {RewardEscrowV2} from "./RewardEscrowV2.sol";
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {ITokenDistributor} from "./interfaces/ITokenDistributor.sol";
+
+contract TokenDistributor is ITokenDistributor {
+    /// @notice represents the status of if a person already
+    /// claimed their epoch
+    mapping(address => mapping(uint => bool)) internal claimedEpochs;
+
+    /// @notice kwenta interface
+    IKwenta private kwenta;
+
+    /// @notice rewards staking contract
+    StakingRewardsV2 private stakingRewardsV2;
+
+    /// @notice escrow contract which holds (and may stake) reward tokens
+    RewardEscrowV2 private rewardEscrowV2;
```

Comment:
Can't really see a reason to make these private, internal is probably good enough in case you want to inherit this contract in the future

Reply:
See [V10](https://github.com/Kwenta/kwenta-solcurity#variables)